source/slang/lower-to-ir.cpp

```diff
@@ -2303,9 +2311,13 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
             ensureDecl(context, inheritanceDecl);
         }
 
-        // For now, we don't have an IR-level representation
-        // for the type itself.
-        return LoweredValInfo();
+        // TODO: we currently store a Decl* in the witness table, which causes this function
+        // being invoked to translate the witness table entry into an IRValue.
+        // We should really allow a witness table entry to represent a type and not having to
```

Comment:
There are a few ways to solve this. One is to have different kinds of witness-table entries, and have associated types use a different one from methods. The other one that might be expedient is to store a `DeclRef` instead of a deck, which should be able to include substitutions.